llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7109,6 +7109,47 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
     }
   }
 
+  {
+    Value *X;
+    Constant *C;
+    if (match(Op0, m_OneUse(m_Intrinsic<Intrinsic::abs>(m_Value(X),
```

Comment:
I think we can drop the one-use constraint as we never create more than one instruction.